addons/spreadsheet/static/src/chart/plugins/odoo_chart_core_plugin.js

```diff
@@ -126,6 +116,10 @@ export class OdooChartCorePlugin extends OdooCorePlugin {
             if (sheet.figures) {
                 for (const figure of sheet.figures) {
                     if (figure.tag === "chart" && figure.data.type.startsWith("odoo_")) {
+                        this.history.update("charts", figure.id, {
+                            id: figure.id,
+                            fieldMatching: figure.data.fieldMatching || {},
+                        });
                         this._addOdooChart(figure.id, figure.data.fieldMatching);
```

Comment:
The call to `_addOdooChart` is useless. Remove it or you can also remove `this.history.update` and add a fallback
`this._addOdooChart(figure.id, figure.data.fieldMatching ?? {});`